### PR TITLE
git_pull: add timestamp to git_pull logs

### PR DESCRIPTION
**Pull request by danielgomezrico**
The logs are like:

```console
and the repository exists.
[Error] Git fetch failed
Check SSH connection
[Warn] No valid SSH connection for git@github.com
[Info] Start adding SSH key
[Info] Setup deployment_key on id_rsa
[Info] Local git repository exists
```

# Expected

They contain the timestamp:
```console
and the repository exists.
[Error][2023-09-26 12:34:56] Git fetch failed
Check SSH connection
[Warn][2023-09-26 12:34:56] No valid SSH connection for git@github.com
[Info][2023-09-26 12:34:56] Start adding SSH key
[Info][2023-09-26 12:34:56] Setup deployment_key on id_rsa
[Info][2023-09-26 12:34:56] Local git repository exists
```

### CODE REVIEW

**Comment by danielgomezrico** [comment]
Im not really sure how to test this... I tried forking, editing and installing the custom repo on my side (https://github.com/danielgomezrico/addons/commits/master) but I get:

```console
Failed to install add-on
Can't install homeassistant/aarch64-addon-git_pull:7.13.2: 404 Client Error for http+docker://localhost/v1.42/images/create?tag=7.13.2&fromImage=homeassistant%2Faarch64-addon-git_pull&platform=linux%2Farm64: Not Found ("manifest for homeassistant/aarch64-addon-git_pull:7.13.2 not found: manifest unknown: manifest unknown")
```

image

**Comment by danielgomezrico** [comment]
Hi @pvizeli, can you take a look? If you can put `hacktoberfest-accepted` label on it, it would be wonderful.

**Comment by agners** [comment]
@danielgomezrico you need to adjust `config.yaml` to point to the location of where you pushed the add-on to.

File: git_pull/data/run.sh [hunk 1]
```diff
@@ -20,16 +20,34 @@ REPEAT_INTERVAL=$(jq --raw-output '.repeat.interval' $CONFIG_PATH)
...
+    log "Error" "$1"
+}
+
+function log_w {
+    log "Warn" "$1"
+}
+
 function add-ssh-key {
-    echo "[Info] Start adding SSH key"
+    log_i "Start adding SSH key"
```
Comment (frenck) [inline git_pull/data/run.sh:42]:
```suggestion
    bashio::log.info "Start adding SSH key"
```

File: git_pull/data/run.sh [hunk 2]
```diff
@@ -63,14 +81,14 @@ function git-clone {
...
+    log_i "Check SSH connection"
     IFS=':' read -ra GIT_URL_PARTS <<< "$REPOSITORY"
     # shellcheck disable=SC2029
     DOMAIN="${GIT_URL_PARTS[0]}"
     if OUTPUT_CHECK=$(ssh -T -o "StrictHostKeyChecking=no" -o "BatchMode=yes" "$DOMAIN" 2>&1) || { [[ $DOMAIN = *"@github.com"* ]] && [[ $OUTPUT_CHECK = *"You've successfully authenticated"* ]]; }; then
-        echo "[Info] Valid SSH connection for $DOMAIN"
+        log_i "Valid SSH connection for $DOMAIN"
     else
-        echo "[Warn] No valid SSH connection for $DOMAIN"
+        log_w "No valid SSH connection for $DOMAIN"
```
Comment (frenck) [inline git_pull/data/run.sh:91]:
```suggestion
        bashio::log.warning "No valid SSH connection for $DOMAIN"
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Pull request closed without merging**